README.md

```diff
@@ -162,24 +162,43 @@ const render = () => html`
 `;
 ```
 
-### Function Values / Directives
+### Directives
 
-A function valued expression can be used for error handling and stateful rendering.
+Directives are functions that can extend lit-html by directly interacting with the Part API.
 
-If an expression returns a function, the function is called with the `Part` its populating, inside a try/catch block.
+Directive will usually be created from factory functions that accept some arguments for values and configuration. Directives are created by passing a function to lit-html's `directive()` function:
 
-This makes it safe from exceptions:
+```javascript
+html`<div>${directive((part) => { part.setValue('Hello')})}</div>`
+```
+
+Here's an example of a directive that takes a function, and evaluates it in a try/catch to implement exception safe expressions:
 
 ```javascript
-let data;
-const render = () => html`foo = ${_=>data.foo}`;
+const safe = (f) => directive((part) => {
+  try {
+    return f();
+  } catch (e) {
+    console.error(e);
+  }
+});
 ```
 
-Here, `data.foo` throws because `data` is undefined, but the rest of the template renders.
+Now `safe()` can be used to wrap a function:
+
+```javascript
+let data;
+const render = () => html`foo = ${safe(_=>data.foo)}`;
+```
 
-And is a useful extension point:
+This example increments a counter on every render:
 
-const render = () => html`<div>${(part) => part.setValue((part.previousValue + 1) || 0)}</div>`;
+```javascript
+const render = () => html`
+  <div>
+    ${directive((part) => part.setValue((part.previousValue + 1) || 0))}
```

Comment:
Couldn't you expand this so that it is clear where part comes from and where previousValue is defined?